microsoft-edge/devtools-guide-chromium/sources/index.md

```diff
@@ -87,49 +256,272 @@ script.integrity = 'sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=';
+*   Press `Ctrl`+`Shift`+`P` (Windows/Linux) or `Command`+`Shift`+`P` (macOS) to open the [Command Menu][DevtoolsGuideChromiumCommandMenuIndex], delete the `>` character, type `!`, type the name of your Snippet, and then select `Enter`.  
+
+Snippets are similar to bookmarklets.
+
+**Related articles:**
+
+*   [Run snippets of JavaScript on any webpage with Microsoft Edge DevTools][DevtoolsGuideChromiumJavascriptSnippets]
+
+## The Editor pane, to view or edit files
+
+Use the **Editor** pane to view JavaScript, HTML, CSS, and other files that are returned from the server.  Make experimental edits to the JavaScript or CSS code returned from the server; DevTools updates the page to run the modified code.  These experimental changes are preserved (in the DevTools sandbox) until page refresh.  
+
+To view and edit local JavaScript, HTML, or CSS source files using the Sources tool, add files to the Workspace, as described above in [The Filesystem tab, for a local Workspace](#the-filesystem-tab-for-a-local-workspace).
+
+:::image type="complex" source="../media/edit-js.msft.png" alt-text="Editing JavaScript in the Editor pane" lightbox="../media/edit-js.msft.png":::
+   Editing JavaScript in the **Editor** pane  
+:::image-end:::
+
+The **Editor** pane has the following level of support for various file types:
+
+| File Type | Supported Actions |
+|---------|---------|
+| JavaScript | View, edit, and debug. |
+| CSS | View and edit. |
+| HTML | View and edit. |
+| Images | View. |
+
+The following subsections are related to the Editor pane:
+*   [Reformat a minified JavaScript file with pretty-print](#reformat-a-minified-javascript-file-with-pretty-print)
+*   [Apply or save changes](#apply-or-save-changes)
+*   [Source mapping](#source-mapping)
+*   [Editor commands on the Command Menu](#editor-commands-on-the-command-menu)
+*   [Edit an HTML file](#edit-an-html-file)
+*   [Edit a CSS file](#edit-a-css-file)
+
+### Reformat a minified JavaScript file with pretty-print
+
+To use pretty-print to reformat a file to make it readable, click the **Pretty print** button \(![Format](../media/format-icon.msft.png)\) at the bottom of the Editor pane.  For more information, navigate to the section
+[Reformat a minified JavaScript file with pretty-print][DevToolsJavaScriptReferenceReformat] within the article "Use the debugger features".
+
+### Apply or save changes
+
+For JavaScript changes to take effect, select `Control`+`S` \(Windows, Linux\) or `Command`+`S` \(macOS\).  DevTools does not re-run a script, so the only JavaScript changes that take effect are those that you make inside of functions.  
+
+For example, in the following figure, notice how `console.log('A')` does not run, whereas `console.log('B')` does.  If DevTools re-runs the entire script after making the change, <!-- why/when would DevTools re-run the entire script?  above says "DevTools does not re-run a script" --> then the text `A` is logged to the **Console**.  
```

Comment:
@zoherghadyali - why/when would DevTools re-run the entire script? above says "DevTools does not re-run a script"